v2/apiserver/internal/core/kubernetes/substrate.go

```diff
@@ -14,11 +14,42 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/selection"
 	"k8s.io/client-go/kubernetes"
 )
 
+var workerPodsSelector = labels.Set(
+	map[string]string{
+		myk8s.LabelComponent: "worker",
+	},
+).AsSelector().String()
+
+var jobPodsSelector = labels.Set(
+	map[string]string{
+		myk8s.LabelComponent: "job",
+	},
+).AsSelector().String()
+
+var runningPodsSelector = fields.Set(
+	map[string]string{
+		"status.phase": string(corev1.PodRunning),
```

Comment:
Is there a typed object for `status.phase`?

Reply:
Good question. I had hoped so, but couldn't find one. A field selector allows arbitrary navigation of nested fields, so I suppose it also makes sense that there wouldn't be.